deps/chakrashim/src/inspector/v8-runtime-agent-impl.cc

```diff
@@ -125,6 +166,29 @@ void V8RuntimeAgentImpl::evaluate(
     return;
   }
 
+  if (isError) {
+    protocol::ErrorSupport errors;
+    std::unique_ptr<protocol::Runtime::RemoteObject> exceptionObject =
+        protocol::Runtime::RemoteObject::parse(protocolValue.get(), &errors);
+    if (!exceptionObject) {
+      errorString = errors.errors();
+      callback->sendFailure(errorString);
+      return;
+    }
+
+    std::unique_ptr<protocol::Runtime::ExceptionDetails> exDetails =
+        protocol::Runtime::ExceptionDetails::create()
+          .setExceptionId(m_session->inspector()->nextExceptionId())
+          .setText(exceptionObject->getDescription("Uncaught"))
+          .setLineNumber(0)
```

Comment:
I'm not clear on what is throwing the exception here. Is this for when the inspector client/user asks for some script to be evaluated, that script throws an uncaught exception, and this is now being reported back to the client/user? Shouldn't these line/column numbers refer to where the exception came from then?

Reply:
You're right, this exception would be thrown due to issues in the evaluated script.  The line and column numbers weren't available so I just defaulted them to zero.  It's probably something we should fix in ChakraCore though.

Reply:
Do you mean like you tried to evaluate some immediately-invalid script, or you evaluated `callSomeFunctionThatThrows()`? I would have expected that if you invoked a function that threw, the site of the throw would have line information associated with it? And even if it was evaluating `throw new Error()` I'd expect that to have some source information, even if it was just treated like an `eval`

Reply:
I'll have to look deeper, but the JsDiagEvaluate function squashes the exception down into a JavaScript object before returning it.  It doesn't necessarily return everything that the exception would have provided.